packages/fx-core/src/plugins/solution/fx-solution/v2/executeUserTask.ts

```diff
@@ -297,9 +299,39 @@ export async function addCapability(
   const alreadyHasTab = solutionSettings.capabilities.includes(TabOptionItem.id);
   const alreadyHasBot = solutionSettings.capabilities.includes(BotOptionItem.id);
   const alreadyHasME = solutionSettings.capabilities.includes(MessageExtensionItem.id);
+  const alreadyHasSso =
+    isAadManifestEnabled() && solutionSettings.capabilities.includes(SsoItem.id);
   const toAddTab = capabilitiesAnswer.includes(TabOptionItem.id);
   const toAddBot = capabilitiesAnswer.includes(BotOptionItem.id);
   const toAddME = capabilitiesAnswer.includes(MessageExtensionItem.id);
+  const toAddTabNonSso = isAadManifestEnabled() && capabilitiesAnswer.includes(TabNonSsoItem.id);
+
+  if (isAadManifestEnabled()) {
+    if (alreadyHasSso && toAddTabNonSso) {
+      const e = new SystemError(
+        SolutionError.InvalidInput,
+        getLocalizedString("core.addSsoFiles.canNotAddNonSsoTabWhenSsoEnabled"),
+        SolutionSource
+      );
+      return err(e);
+    }
+
+    if (!alreadyHasSso && toAddTab) {
+      const e = new SystemError(
+        SolutionError.InvalidInput,
+        getLocalizedString("core.addSsoFiles.canNotAddTabWhenSsoNotEnabled"),
+        SolutionSource
+      );
+      return err(e);
+    }
+
+    if (toAddTabNonSso) {
+      const index = capabilitiesAnswer.indexOf(TabNonSsoItem.id);
+      capabilitiesAnswer.splice(index);
```

Comment:
`Array.splice()` here will remove all items within range [index, capabilitiesAnswer.length).

Reply:
Besides, I'm a little bit confused about why you want to remove `TabNonSsoItem.id` and manually add `TabOptionItem.id` here when `toAddTabNonSso` is true.

Reply:
Both tab and non sso tab activates fe plugin, and fe plugin will handle the logic whether to scaffold sso or non sso template. What we need to do is to add a tab. By remove the non sso tab and add the tab option, we can keep the logic below same as current :
![image](https://user-images.githubusercontent.com/63089166/160087430-853ae56f-8e4a-498a-8589-d5fc81ae5175.png)

Reply:
OK, that's a little bit tricky.